.github/workflows/pr-github-checks.yml

```diff
@@ -70,10 +70,46 @@ jobs:
           requireScope: false
           subjectPattern: ^([A-Z].*[^.]|bump .*)$
 
+  pr-prevent-kustomization:
+    runs-on: ubuntu-latest
+    if: ${{ github.event.pull_request.base.ref == 'main' }}
+    steps:
+      - name: Check-out code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Check kustomization.yaml changes
+        uses: tj-actions/changed-files@6b2903bdce6310cfbddd87c418f253cf29b2dec9
+        id: changed-kustomization
+        with:
+          files: "config/manager/kustomization.yaml"
+      - name: Prevent kustomization.yaml changes
+        run: |
+          if ${{ steps.changed-kustomization.outputs.any_changed }}; then
+            echo "kustomization.yaml file changed!"
+            exit 1
+          fi
+
   PR-Github-Checks-Success:
-    needs: [pr-milestone-check, pr-label-check, pr-title-check]
+    needs:
+      [
+        pr-milestone-check,
+        pr-label-check,
+        pr-title-check,
+        pr-prevent-kustomization,
+      ]
     runs-on: ubuntu-latest
+    if: always()
     steps:
+      - name: PR Github Checks
+        run: |
+          if ! ${{ needs.pr-milestone-check.result == 'success' }} &&
+          ${{ needs.pr-label-check.result == 'success' }} &&
+          ${{ needs.pr-title-check.result == 'success' }} &&
+          (${{ needs.pr-prevent-kustomization.result == 'success' }} || ${{ needs.pr-prevent-kustomization.result == 'skipped' }}); then
```

Comment:
we skip the kustomization check if branch is not main. So this would cause the github check to fail for `release` branch ?

Reply:
No, because for the pr-prevent-kustomization job I also treat the skipped scenario as a positive one, see line 109